src/client.ts

```diff
@@ -76,7 +78,151 @@ export default function connect(
   dsn?: string | ConnectConfig | null,
   options?: ConnectConfig | null
 ): Promise<Connection> {
-  return ConnectionImpl.connect(dsn, options);
+  let config: ConnectConfig | null = null;
+  if (typeof dsn === "string") {
+    config = {...options, dsn};
+  } else {
+    if (dsn != null) {
+      // tslint:disable-next-line: no-console
+      console.warn(
+        "`options` as the first argument to `edgedb.connect` is " +
+          "deprecated, use " +
+          "`edgedb.connect('instance_name_or_dsn', options)`"
+      );
+    }
+    config = {...dsn, ...options};
+  }
+  return StandaloneConnection.connect(parseConnectArguments(config));
+}
+
+function sleep(durationMillis: number): Promise<void> {
+  return new Promise((accept, reject) => {
+    setTimeout(() => accept(), durationMillis);
+  });
+}
+
+class StandaloneConnection implements Connection {
+  [ALLOW_MODIFICATIONS]: never;
+  private config: NormalizedConnectConfig;
+  private _connection?: ConnectionImpl;
+  private _isClosed: boolean; // For compatibility
+
+  private constructor(config: NormalizedConnectConfig) {
+    this.config = config;
+    this._isClosed = false;
+  }
+  async transaction<T>(
+    action: () => Promise<T>,
+    options?: TransactionOptions
+  ): Promise<T> {
+    let connection = this._connection;
+    if (!connection || connection.isClosed()) {
+      connection = await this._reconnect();
+    }
+    return await connection.transaction(action, options);
+  }
+  async close(): Promise<void> {
+    try {
+      if (this._connection) {
+        await this._connection.close();
+      }
+      this._connection = undefined;
+      // TODO(tailhook) it makes little sense to close the reconnecting
+      // connection so maybe deprecate this method
```

Comment:
Not sure I follow this TODO comment. Can you elaborate?

Reply:
Currently:
```
await conn.close()
await conn.query('SELECT 1`)
```
Works fine, as we reconnect if connection was closed (i.e. when it is closed because of network error).

As it still closes underlying connection it may be helpful, though. So yeah, perhaps this comment is useless.

Reply:
> Works fine, as we reconnect if connection was closed (i.e. when it is closed because of network error).

But it shouldn't. If the connection was explicitly closed by user it should not reconnect, it should be marked as closed and all operations on it should be rejected. It's a basic contract that has existed since forever and I don't think we are ought to break it.